src/index.js

```diff
@@ -207,28 +207,23 @@ export default function undoable (reducer, rawConfig = {}) {
     clearHistoryType: rawConfig.clearHistoryType || ActionTypes.CLEAR_HISTORY
   }
 
-  return (state, action = {}) => {
+  return (state = config.history, action = {}) => {
     debugStart(action, state)
 
-    let history
+    let history = state
     if (!config.history) {
-      debug('create history on init')
+      debug('history is uninitialized')
 
       if (state === undefined) {
-        config.history = createHistory(reducer(state, {}))
+        history = createHistory(reducer(state, {}))
+        debug('do not initialize on probe actions')
       } else if (isHistory(state)) {
-        config.history = state
+        history = config.history = state
+        debug('initialHistory properly shaped and initialized', config.history)
       } else {
-        config.history = createHistory(state)
+        history = config.history = createHistory(state)
+        debug('initialHistory reshaped and initialized', config.history)
```

Comment:
you could consider rephrasing this to "create history from initialState". It's not really re-shaping what you're doing here right?